### ISSUE TITLE
`type[X] | type[Y]` -> `type[X | Y]`

### ISSUE THREAD

**Issue by AlexWaygood** [PyCQA/flake8-pyi#175]
And, similarly, `Union[type[X], type[Y]]` -> `type[Union[X, Y]]`.

Same basic idea as #110. Came up in https://github.com/python/typeshed/pull/7087.

**Comment by tomasr8** [comment]
Hi! Is this issue still relevant? If so, I'd like to give it a try :)

**Comment by AlexWaygood** [comment]
> Hi! Is this issue still relevant? If so, I'd like to give it a try :)

Yes, very much so — feel free! I took a quick look at it some time ago and remember it being slightly trickier than I anticipated — but the codebase has changed a lot since, so that won't necessarily still hold true :)
